src/calcwit.cpp

```diff
@@ -65,6 +65,20 @@ Circom_CalcWit::Circom_CalcWit (Circom_Circuit *aCircuit, uint maxTh) {
 
 Circom_CalcWit::~Circom_CalcWit() {
   // ...
+  if (inputSignalAssigned != nullptr) {
+    delete[] inputSignalAssigned;
+    inputSignalAssigned = nullptr;
+  }
```

Comment:
It is not necessary to check for nullptr before delete[] and assign nullptr after.
The same in other places.

Reply:
Thank you for the reminder. Initially, I also planned not to check for nullptr, but I noticed that the code in authv2.app does perform such checks. 

```cpp
if(ctx->componentMemory[pos].subcomponents)
delete []ctx->componentMemory[pos].subcomponents;
```

Therefore, I intend to adopt the same coding style. I will remove these conditional checks immediately."